presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerIntegrationSmokeTest.java

```diff
@@ -98,14 +100,40 @@ public void testAggregationPushdown()
             assertThat(query("SELECT sum(short_decimal), sum(long_decimal) FROM test_aggregation_pushdown")).isCorrectlyPushedDown();
             assertThat(query("SELECT avg(short_decimal), avg(long_decimal) FROM test_aggregation_pushdown")).isCorrectlyPushedDown();
+
+            // smoke testing of more complex cases
+            // WHERE on aggregation column
+            assertThat(query("SELECT min(short_decimal), min(long_decimal) FROM test_aggregation_pushdown WHERE short_decimal < 110 AND long_decimal < 124")).isCorrectlyPushedDown();
+            // WHERE on non-aggregation column
+            assertThat(query("SELECT min(long_decimal) FROM test_aggregation_pushdown WHERE short_decimal < 110")).isCorrectlyPushedDown();
+            // GROUP BY
+            assertThat(query("SELECT short_decimal, min(long_decimal) FROM test_aggregation_pushdown GROUP BY short_decimal")).isCorrectlyPushedDown();
+            // GROUP BY with WHERE on both grouping and aggregation column
+            assertThat(query("SELECT short_decimal, min(long_decimal) FROM test_aggregation_pushdown WHERE short_decimal < 110 AND long_decimal < 124" + " GROUP BY short_decimal")).isCorrectlyPushedDown();
+            // GROUP BY with WHERE on grouping column
+            assertThat(query("SELECT short_decimal, min(long_decimal) FROM test_aggregation_pushdown WHERE short_decimal < 110 GROUP BY short_decimal")).isCorrectlyPushedDown();
+            // GROUP BY with WHERE on aggregation column
+            assertThat(query("SELECT short_decimal, min(long_decimal) FROM test_aggregation_pushdown WHERE long_decimal < 124 GROUP BY short_decimal")).isCorrectlyPushedDown();
+            // GROUP BY with WHERE on neither grouping nor aggregation column
+            assertThat(query("SELECT short_decimal, min(long_decimal) FROM test_aggregation_pushdown WHERE varchar_column = 'ala' GROUP BY short_decimal")).isCorrectlyPushedDown();
+            // aggregation on varchar column
+            assertThat(query("SELECT min(varchar_column) FROM test_aggregation_pushdown")).isCorrectlyPushedDown();
+            // aggregation on varchar column with GROUPING
+            assertThat(query("SELECT short_decimal, min(varchar_column) FROM test_aggregation_pushdown GROUP BY short_decimal")).isCorrectlyPushedDown();
+            // aggregation on varchar column with WHERE
+            assertThat(query("SELECT min(varchar_column) FROM test_aggregation_pushdown WHERE varchar_column ='ala'")).isCorrectlyPushedDown();
+
+            // not supported yet
+            assertThat(query("SELECT min(DISTINCT short_decimal) FROM test_aggregation_pushdown")).isNotFullyPushedDown(AggregationNode.class);
+            assertThat(query("SELECT DISTINCT short_decimal, min(long_decimal) FROM test_aggregation_pushdown GROUP BY short_decimal")).isNotFullyPushedDown(ProjectNode.class);
```

Comment:
why ProjectNode and not, say, AggregationNode?

Reply:
This requires explanation. In other (similar) test I put following comment by this line:

```
// TODO: Improve assertion framework. Here min(long_decimal) is pushed down. There remains ProjectNode above it which relates to DISTINCT in the query.
```

Maybe `isCorrectlyPushedDown` should allow for extra `ProjectNode`?

Reply:
Or we can add `isCorrectlyPushedDown(Class<? extends PlanNode> allowedRetainedNodes)`

Reply:
i got the plan:

```
presto:dbo> explain SELECT DISTINCT short_decimal, min(long_decimal) FROM test_aggregation_pushdown GROUP BY short_decimal;
                                                                                                                                         Query Plan
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Fragment 0 [SINGLE]
     Output layout: [short_decimal, _presto_generated]
     Output partitioning: SINGLE []
     Stage Execution Strategy: UNGROUPED_EXECUTION
     Output[short_decimal, _col1]
     │   Layout: [short_decimal:decimal(9,3), _presto_generated:decimal(30,10)]
     │   Estimates: {rows: ? (?), cpu: ?, memory: ?, network: ?}
     │   _col1 := _presto_generated
     └─ RemoteSource[1]
            Layout: [short_decimal:decimal(9,3), _presto_generated:decimal(30,10)]

 Fragment 1 [HASH]
     Output layout: [short_decimal, _presto_generated]
     Output partitioning: SINGLE []
     Stage Execution Strategy: UNGROUPED_EXECUTION
     Project[]
     │   Layout: [short_decimal:decimal(9,3), _presto_generated:decimal(30,10)]
     │   Estimates: {rows: ? (?), cpu: ?, memory: ?, network: ?}
     └─ Aggregate(FINAL)[short_decimal, _presto_generated][$hashvalue]
        │   Layout: [short_decimal:decimal(9,3), _presto_generated:decimal(30,10), $hashvalue:bigint]
        │   Estimates: {rows: ? (?), cpu: ?, memory: ?, network: ?}
        └─ LocalExchange[HASH][$hashvalue] ("short_decimal", "_presto_generated")
           │   Layout: [short_decimal:decimal(9,3), _presto_generated:decimal(30,10), $hashvalue:bigint]
           │   Estimates: {rows: ? (?), cpu: ?, memory: ?, network: ?}
           └─ RemoteSource[2]
                  Layout: [short_decimal:decimal(9,3), _presto_generated:decimal(30,10), $hashvalue_0:bigint]

 Fragment 2 [SOURCE]
     Output layout: [short_decimal, _presto_generated, $hashvalue_1]
     Output partitioning: HASH [short_decimal, _presto_generated][$hashvalue_1]
     Stage Execution Strategy: UNGROUPED_EXECUTION
     Aggregate(PARTIAL)[short_decimal, _presto_generated][$hashvalue_1]
     │   Layout: [short_decimal:decimal(9,3), _presto_generated:decimal(30,10), $hashvalue_1:bigint]
     └─ ScanProject[table = sqlserver:dbo.test_aggregation_pushdown master.dbo.test_aggregation_pushdown columns=[short_decimal:decimal(9,3):decimal, min("long_decimal"):_presto_generated_1:decimal(30,10):decimal] groupingSets=[[short_decimal:decimal(9,3):decimal]], group
            Layout: [short_decimal:decimal(9,3), _presto_generated:decimal(30,10), $hashvalue_1:bigint]
            Estimates: {rows: ? (?), cpu: ?, memory: 0B, network: 0B}/{rows: ? (?), cpu: ?, memory: 0B, network: 0B}
            $hashvalue_1 := combine_hash(combine_hash(bigint '0', COALESCE("$operator$hash_code"("short_decimal"), 0)), COALESCE("$operator$hash_code"("_presto_generated"), 0))
            short_decimal := short_decimal:decimal(9,3):decimal
            _presto_generated := min("long_decimal"):_presto_generated_1:decimal(30,10):decimal
```

i see Agg is not pushed down. 

what plan do you get here?

Reply:
This is different Agg, right? This is distinct converted to aggregation.
The `min` is being pushed down. 

IDK how to cover it accurately without writing matcher for whole expected plan.

Reply:
The point is -- there are two aggregations, one on top of the other.

`isNotFullyPushedDown(AggregationNode)` would be correct, since the upper agg is indeed not pushed

we currently cannot express the fact that the lower one _is_ pushed, but maybe we can assume it's mostly covered by simpoler test cases

Reply:
Sure. I can change the extra expected node to `AggregationNode`. But I will still leave the comment to explain what is happening there.

Reply:
I take that back. I cannot put `AggregateNode` there.
As `isNotFullyPushedDown` then expectes a plan like that:
```
- anyTree
    - node(AggregationNode)
        - node(TableScanNode)
```
And we also have an extra `ProjectNode` below `Aggregation`.